logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/SendEditTextMessageUseCase.kt

```diff
@@ -83,32 +83,46 @@ class SendEditTextMessageUseCase internal constructor(
                 newMentions = mentions
             )
             val message = Message.Signaling(
-                id = generatedMessageUuid,
+                id = editedMessageId,
                 content = content,
                 conversationId = conversationId,
                 date = DateTimeUtil.currentIsoDateTimeString(),
                 senderUserId = selfUserId,
                 senderClientId = clientId,
                 status = Message.Status.PENDING
             )
+            // until the edit send is completed and accepted by the backend, we don't change the message id to be able to handle any
+            // incoming edits from other clients that happened in the meantime and already changed the message id
             messageRepository.updateTextMessage(
                 conversationId = message.conversationId,
                 messageContent = content,
-                newMessageId = generatedMessageUuid,
+                newMessageId = originalMessageId,
                 editTimeStamp = message.date
             ).flatMap {
-                messageRepository.updateMessageStatus(
-                    messageStatus = MessageEntity.Status.PENDING,
-                    conversationId = message.conversationId,
-                    messageUuid = generatedMessageUuid
-                )
-            }.flatMap {
-                messageSender.sendMessage(message)
-            }.flatMap {
-                messageRepository.updateMessageStatus(MessageEntity.Status.SENT, conversationId, generatedMessageUuid)
-            }
+                    messageRepository.updateMessageStatus(
+                        messageStatus = MessageEntity.Status.PENDING,
+                        conversationId = message.conversationId,
+                        messageUuid = originalMessageId
+                    )
+                }
+                .flatMap {
+                    messageSender.sendMessage(message)
+                        .flatMap {
+                            // when succeeded, we modify the message id locally
+                            messageRepository.updateTextMessage(
+                                conversationId = message.conversationId,
+                                messageContent = content,
+                                newMessageId = editedMessageId,
+                                editTimeStamp = message.date
+                            )
+                        }
+                        .flatMap {
+                            // and change the status of the message
+                            messageRepository.updateMessageStatus(MessageEntity.Status.SENT, conversationId, editedMessageId)
```

Comment:
I think we also need to do this in the `MessageTextEditHandler` since it might be pending

Reply:
I already added that:
https://github.com/wireapp/kalium/pull/1583/files#diff-dad9f1e934e93a119a9e21d403005607074ed87f1f1507e2ffe2ebc8fdfae896R82